src/traces/scattergl/line_shape_defaults.js

```diff
@@ -0,0 +1,15 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+module.exports = function handleLineShapeDefaults(traceIn, traceOut, coerce) {
+    var shape = coerce('line.shape');
```

Comment:
No need for a new file here.

Putting that `coerce('line.shape')` in the `if(subTypes.hasLines(traceOut))` block will suffice.

Reply:
last version of commit resolve this